frontend/apps/filemanager/filemanager.lua

```diff
@@ -433,6 +433,7 @@ function FileManager:init()
         logger.err("FileManager instance mismatch! Opened", tostring(self), "while we still have an existing instance:", tostring(FileManager.instance), debug.traceback())
     end
     FileManager.instance = self
+    G_ui = self
```

Comment:
Haven't looked at the full picture yet, but, yep, looks nasty and potentially highly error-prone ;).

Reply:
Yep, I hate it ;).

Globals are evil to begin with, and this pins the whole subtree in a global, making it doubly so ;).

Reply:
What is the negative impact, technically?

Reply:
Globals are bad form. Globals are also costly.

This also pins the *whole* tree in memory, because it's the root of basically all of our things, so it has at least one ref of basically everything.

Add to that that the GC is extremely cautious about collecting globals, IIRC.

That makes for a terrible combination ;).